src/graphql/triage/policy.rs

```diff
@@ -71,36 +91,129 @@ async fn load(
+        policies.reverse();
+        (policies, has_more, false)
+    } else {
+        let first = first.unwrap_or(DEFAULT_CONNECTION_SIZE);
+        let (policies, has_more) = iter_to_policies(
+            &map,
+            Direction::Forward,
+            after.as_deref(),
+            before.as_deref(),
+            cmp::Ordering::is_le,
+            first,
+            customer_id,
+        )?;
+        (policies, false, has_more)
+    };
+
+    let mut connection = Connection::with_additional_fields(
+        has_previous,
+        has_next,
+        TriagePolicyTotalCount { customer_id },
+    );
+    connection.edges.extend(
+        policies
+            .into_iter()
+            .map(|(key, policy)| Edge::new(OpaqueCursor(key), policy)),
+    );
+    Ok(connection)
+}
+
+fn iter_to_policies(
+    map: &database::IndexedTable<'_, database::TriagePolicy>,
+    direction: Direction,
+    start: Option<&[u8]>,
+    bound: Option<&[u8]>,
+    cond: fn(cmp::Ordering) -> bool,
+    len: usize,
+    customer_id: Option<u32>,
+) -> anyhow::Result<(Vec<PolicyEdge>, bool)> {
+    let mut policies = Vec::new();
+    let mut exceeded = false;
+    let mut iter = map.iter(direction, start);
+    // exclusive start cursor: skip the first item when a start is provided
+    if start.is_some() {
+        iter.next();
+    }
+    for item in iter {
+        let policy = item?;
+        let key = policy.unique_key();
+
+        if let Some(b) = bound
+            && !(cond)(key.as_slice().cmp(b))
+        {
+            break;
+        }
+        if !matches_customer(&policy, customer_id) {
+            continue;
+        }
+        policies.push((key, policy.into()));
+        exceeded = policies.len() > len;
+        if exceeded {
+            break;
+        }
+    }
+    if exceeded {
+        policies.pop();
+    }
+    Ok((policies, exceeded))
+}
+
+// Matches a policy against the caller's customer scope:
+//
+// customer_id Some: allow global policies (customer None) or policies for that specific customer.
+// customer_id None: System administrator context, so all policies are visible.
+fn matches_customer(policy: &database::TriagePolicy, customer_id: Option<u32>) -> bool {
+    match customer_id {
+        Some(id) => policy.customer_id.is_none() || policy.customer_id == Some(id),
+        None => true,
+    }
 }
 
 #[Object]
 impl TriagePolicyMutation {
     /// Inserts a new triage policy, returning the ID of the new triage.
+    #[allow(clippy::too_many_arguments)]
     #[graphql(guard = "RoleGuard::new(Role::SystemAdministrator)
         .or(RoleGuard::new(Role::SecurityAdministrator))")]
     async fn insert_triage_policy(
         &self,
         ctx: &Context<'_>,
         name: String,
-        ti_db: Vec<TriageExclusionReasonInput>,
+        triage_exclusion_id: Vec<ID>,
         packet_attr: Vec<PacketAttrInput>,
         confidence: Vec<ConfidenceInput>,
         response: Vec<ResponseInput>,
+        customer_id: Option<ID>,
```

Comment:
It looks like that the method validates `customer_id` can be successfully parsed but doesn't check whether it exists. Is it good?

Reply:
I don't understand. This API is one of the GraphQL APIs accessible to any users. How can we make it sure that the input is always legitimate?